src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -89,6 +97,22 @@ public void addMessage(Message message) {
     persistentStorageAgent.writeThrough(message);
   }
 
+  /** Delete a message from the current set of messages known to the application */
+  public void deleteMessage(Message message) {
+    messages.remove(message);
+    persistentStorageAgent.delete(message);
+  }
+
+  /** Delete an old message sent by this User */
+  public void deleteOldMessages(UUID userId) {
+    if (messagesByAuthorId.containsKey(userId)) {
+      List<Message> messagesByAuthor = messagesByAuthorId.get(userId);
```

Comment:
This doesn't guarantee the messages are sorted from oldest to newest, I'd try and do that to ensure oldest messages are deleted first.

I'd use the getMessagesByAuthor function below as it sorts the messages. (The compareTo method puts them ascending order, or oldest to newest)